src/Merchants/Content/Merchant/Services/CustomerSync.php

```diff
@@ -0,0 +1,151 @@
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWrittenEvent;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CustomerSync implements EventSubscriberInterface
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $merchantRepository;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(
+        EntityRepositoryInterface $customerRepository,
+        EntityRepositoryInterface $merchantRepository,
+        Connection $connection
+    ) {
+        $this->customerRepository = $customerRepository;
+        $this->merchantRepository = $merchantRepository;
+        $this->connection = $connection;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            'merchant.written' => 'syncToCustomer',
+        ];
+    }
+
+    public function syncToCustomer(EntityWrittenEvent $event) :void
+    {
+        if(count($event->getErrors())) {
+            return;
+        }
+
+        $saneDefaults = [
+            'groupId' => Defaults::FALLBACK_CUSTOMER_GROUP,
+            'defaultPaymentMethodId' => $this->fetchRandomPaymentMethodId(),
+            'languageId' => Defaults::LANGUAGE_SYSTEM,
+            'defaultBillingAddress' => [
+                'countryId' => $this->fetchRandomCountry(),
+                'salutationId' => $this->fetchUnspecifiedSalutation(),
+                'firstName' => '&nbsp;',
+                'lastName' => '&nbsp;',
+                'zipcode' => '123465',
+                'city' => 'ABC',
+                'street' => 'Sesamestreet'
+            ],
+            'defaultShippingAddress' => [
+                'countryId' => $this->fetchRandomCountry(),
+                'salutationId' => $this->fetchUnspecifiedSalutation(),
+                'firstName' => '&nbsp;',
+                'lastName' => '&nbsp;',
+                'zipcode' => '123465',
+                'city' => 'ABC',
+                'street' => 'Sesamestreet'
+            ],
+            'salutationId' => $this->fetchUnspecifiedSalutation(),
+        ];
+
+        $merchantUpdate = [];
+
+        $customers = [];
+        foreach($event->getWriteResults() as $writeResult) {
+            $customer = [$this->extractValuesIfPresent($writeResult->getPayload(), 'salesChannelId', 'name', 'email', 'password')];
+
+            if($customer === [[]]) {
+                continue;
+            }
+
+            if(!$writeResult->getExistence()->exists()) {
+                $newCustomerId = Uuid::randomHex();
+
+                $customer[] = $saneDefaults;
+                $customer[] = [
+                    'id' => $newCustomerId,
+                    'customerNumber' => Uuid::randomHex(),
```

Comment:
Do we not need to save also the password to the customer. So we can login as him?

Reply:
Maybe.... ;D Currently there is not test for that.